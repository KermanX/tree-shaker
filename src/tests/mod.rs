use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
