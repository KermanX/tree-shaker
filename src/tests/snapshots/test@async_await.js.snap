---
source: src/tests/mod.rs
assertion_line: 32
expression: tree_shake(input)
input_file: src/tests/fixtures/async_await.js
---
export const l1 = (await 1, 1);
export const l2 = (await 1, "1a");
export const l3 = await unknown;
export const l4 = await (unknown + "a");
async function pure() {
	return 1;
}
export const r1 = await pure();
export const r2 = pure();
async function nested_pure() {
	return pure();
}
export const r3 = await nested_pure();
async function effect() {
	await something;
	return 1;
}
export const r4 = await effect();
function effect2() {
	return something;
}
export const r5 = await effect2();
export async function pure_but_complex(unknown) {
	const f = async () => unknown;
	return await f();
}
